events/api.py

```diff
@@ -3618,11 +3627,39 @@ class EventFilter(django_filters.rest_framework.FilterSet):
         widget=django_filters.widgets.CSVWidget(),
         method=partial(in_or_null_filter, "super_event"),
     )
+    dwithin = django_filters.Filter(
+        method="filter_dwithin", widget=DistanceWithinWidget()
+    )
 
     class Meta:
         model = Event
         fields = ("division", "super_event_type", "super_event")
 
+    @staticmethod
+    def filter_dwithin(queryset, name, value: tuple[tuple[str, str], str]):
+        origin, metres = value
+        if not (origin and metres):
+            # Need both values for filtering
+            return queryset
+
+        try:
+            origin_x, origin_y = [float(i) for i in origin]
+        except ValueError:
+            raise ParseError("Origin must be a tuple of two numbers")
+        try:
+            metres = float(metres)
+        except ValueError:
+            raise ParseError("Metres must be a number")
+
+        places = Place.geo_objects.filter(
+            position__distance_lte=(
```

Comment:
IMO the filter should not be named `dwithin` when it actually does not use the [DWithin SQL function](https://postgis.net/docs/ST_DWithin.html) 🙂 `distance_lt` would be consistent with at least some of the other filters (`audience_max_age_lt`).

Reply:
Right, not sure why I used `distance_lte` in the first place since I was going to use `dwithin` 😅 well spotted

Reply:
Kept the name, but changed the query to use `dwithin` instead of `distance_lte`